lighthouse-core/gather/gatherers/dobetterweb/links.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+function getPageLinks() {
+  return new Promise((resolve, reject) => {
+    const linkList = [...document.querySelectorAll('a')]
+        .map((link) => {
+          return {
+            target: link.getAttribute('target'),
+            rel: link.getAttribute('rel')
+          };
+        });
+    resolve(linkList);
+  });
+}
+
+class Links extends Gatherer {
+
+  afterPass(options) {
+    const driver = options.driver;
+    const scriptStr = `(${getPageLinks.toString()}())`;
+    return driver.evaluateAsync(scriptStr)
```

Comment:
I tried writing the driver for that yesterday but couldn't get it together because there were some unknowns to me. I've written it now and moved the implementation to use that instead of `evaluateAsync`.